schemes/scheme-model.js

```diff
@@ -0,0 +1,62 @@
+const db = require('../data/db-config');
+
+function find() {
+	return db('schemes')
+		.select();
+}
+
+function findById(id) {
+	return db('schemes')
+		.where({ id })
+		.first();
+};
+
+function findSteps(id) {
+	return db('steps as st')
+		.join('schemes as s', 's.id', 'st.scheme_id')
+		.where({ scheme_id:id })
+		.select('st.id', 's.scheme_name', 'st.step_number', 'st.instructions')
+		.orderBy('st.step_number', 'asc')
+};
+
+async function add(scheme) {
+	const [id] = await db('schemes').insert(scheme)
+	return db('schemes')
+		.where({ id })
+		.first()
+};
+
+async function update(changes, id) {
+	await db('schemes')
+		.where({ id })
+		.update(changes)
+	
+		return findById(id)
+};
+
+function remove(id) {
+	return db('schemes')
+		.where({ id })
+		.select()
+		.then(scheme => {
+			if (!scheme) {
+				return null
+			} else {
+				return db('schemes')
+					.where({ id })
+					.del()
+					.then(() => {
+						return scheme;
+					})
+			}
+		})
+};
```

Comment:
You code looks functional, but I might make some changes.  First you're using async above but then/catch here. I would stick to one or another.

So I would do:
```
async function remove(id) {
  const scheme = await findById(id);
  await db("schemes")
    .where({ id })
    .del();
  return scheme;
}

function remove2(id) {
  db.findById(id).then(scheme => {
    db("schemes")
      .where({ id })
      .del()
      .then(() => scheme);
  });
}
```